blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/AstyanaxReader.java

```diff
@@ -559,4 +630,53 @@ public Boolean apply(Row<Locator, Long> row) {
         return excessEnumMetrics;
     }
 
+    public <T extends Rollup> Points<T> transformEnumValueHashesToStrings(MetricData metricData, ColumnList<Long> enumvalues) {
+        Map<Long, String> hash2enumValues = getEnumValueFromHashes(enumvalues);
+        Points<T> pointsEnum = new Points<T>();
+
+        Map<Long, Points.Point<T>> pointsMap = metricData.getData().getPoints();
+
+        for (Long timestamp : pointsMap.keySet()) {
+            BluefloodEnumRollup enumRollup = (BluefloodEnumRollup)pointsMap.get(timestamp).getData();
+            for (Long hash : enumRollup.getHashedEnumValuesWithCounts().keySet()) {
+                String enumValueString = hash2enumValues.get(hash);
+                enumRollup.getStringEnumValuesWithCounts().put(enumValueString, enumRollup.getHashedEnumValuesWithCounts().get(hash));
+                pointsEnum.add(new Points.Point<T>(timestamp,(T)enumRollup));
+            }
+        }
+        return pointsEnum;
+    }
+
+    private Map<Long, String> getEnumValueFromHashes(ColumnList<Long> enumValues) {
+        HashMap<Long,String> hash2enumValues = new HashMap<Long, String>();
+
+        for (Column<Long> col: enumValues) {
+            hash2enumValues.put(col.getName(), col.getStringValue());
+        }
+
+        return hash2enumValues;
+    }
+
+    public Map<Locator, ColumnList<Long>> getStringAndHashMappingForEnums(final List<Locator> locators) {
```

Comment:
NIT:  Looking at this method, it looks like you're only get querying for HashMapping for the locators passed in.  I am not sure what the "String" part in the method name "getStringAndHashMappingForEnums" refer to.

I would just go with "getEnumHashMappings(final List<Locator> locators)", but no biggie if you disagree.